books/spiders/book.py

```diff
@@ -0,0 +1,56 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BookSpider(scrapy.Spider):
+    name = "book"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs) -> None:
+        for product_url in response.css("h3 > a::attr(href)").getall():
+            product_url = response.urljoin(product_url)
+            yield scrapy.Request(product_url, callback=self.parse_product)
+
+        next_page = response.css("li.next > a::attr(href)").get()
+
+        if next_page:
+            yield response.follow(next_page, callback=self.parse)
+
+    def extract_book_name(self, response: Response) -> str:
+        return response.css("h1::text").get()
+
+    def get_price(self, response: Response) -> int:
+        return response.css(".price_color::text").get()
+
+    def get_amount_in_stock(self, response: Response) -> int:
+        return int(
+            response.css(
+                ".table.table-striped tr td::text"
+            ).getall()[-2].split()[-2].replace("(", "")
+        )
+
+    def get_rating(self, response: Response) -> int:
+        return response.css("p.star-rating::attr(class)").get().split()[-1]
```

Comment:
The `get_rating` method returns a string representing the rating class. If you intend to convert this to a numerical rating, you should map the class name to a corresponding integer value.